### DISCUSSION TITLE
Roborock S5 updated but no web access

### DISCUSSION THREAD

**Discussion by diasdmhub** [Hypfer/Valetudo#2126] [Q&A / Support]
### Are you already using Valetudo?

No, but in the process of starting it

> Version before the update: `3.3.9_001886`

### Your questions

I tried to follow the [Roborock OTA guide](https://valetudo.cloud/pages/installation/roborock.html#ota) and sort of managed to push its firmware to my Roborock S5. However, I can't access the device through a browser.

### Here are the steps I took.

1. First, I reset the vacuum as instructed. The device also reported that it was restoring its firmware.

2. Then I went to DustBuilder and created a Roborock S5 firmware update package. It was sent to my e-mail.

3. Back to the guide, I downloaded the Valetudo Helper MiioOTA tool.

4. I searched for the vacuum Wi-Fi network and connected to it. The `192.168.8.1` address was responsive.

5. Finally, I started Valetudo Helper and it seems that the firmware was successfully pushed to the vacuum. I also left the PC right next to the vacuum.

```bash
$ ./valetudo-helper-miioota-amd64 install-firmware v11_002034.pkg 
Starting installer.
If you experience issues, make sure to disable your firewall and/or VPN.
Also, make sure that the robot is docked during the firmware update procedure.
If the install still fails, try turning the robot off and back on again and/or moving the laptop closer to it.

Robot discovery started...
Scan done.
Successfully discovered robot at 192.168.8.1
Reading firmware image..
Successfully read firmware image. Size: 77.51 MiB MD5Sum: 2440f5a347e39ff62a2d928c30d85649

Listing for firmware download requests on http://192.168.8.54:40811/firmware
Response from robot: [ 'ok' ]
Received firmware download request from ::ffff:192.168.8.1..

Download seems to have finished.
The robot should now install the firmware. It will take 5-10 minutes.

	If you like this application, you may want to consider donating:
	https://github.com/sponsors/Hypfer
Exiting..
```

After all this, nothing else happened and the vacuum remains inaccessible on port 80, as well as SSH.
I tried rebooting it after that, but nothing.

```
$ nc -zv 192.168.8.1 80
nc: connect to 192.168.8.1 port 80 (tcp) failed: Connection refused
```

I'm not sure if I did something wrong or the problem is something else.
Any thoughts on this?

**Comment by Hypfer** [comment]
Usually, this is caused by the robot not being docked while trying to install the firmware.

The download finishes but it just doesn't install it.

**Reply by diasdmhub** [reply]
Hey. Thanks for the input.

When I  tried to update the vacuum out of the dock, it issued a message saying that it should be "charged" before the update, even though it is.

However, I did the procedure with the vacuum in the dock and nothing happened.

**Comment by diasdmhub** [comment]
I was literally capturing packets to try to find the problem.

I noticed many TCP retransmissions. So, it seems there was too much packet loss between the PC and the vacuum.
After about 10 more attempts, it finally worked and the firmware was updated.

I must say that I'm impressed with how nice Valetudo software is.
Thanks a lot guys.
